app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ActionExecutionResult.java

```diff
@@ -38,9 +39,16 @@
     public void setErrorInfo(Throwable error) {
         this.body = error.getMessage();
 
-        if (error instanceof BaseException) {
+        if (error instanceof AppsmithPluginException) {
+            this.statusCode = ((AppsmithPluginException) error).getAppErrorCode().toString();
+            this.title = ((AppsmithPluginException) error).getTitle();
+            this.errorType = ((AppsmithPluginException) error).getErrorType();
+        }
+
+        else if (error instanceof BaseException) {
             this.statusCode = ((BaseException) error).getAppErrorCode().toString();
             this.title = ((BaseException) error).getTitle();
+            this.errorType = ((AppsmithPluginException) error).getErrorType();
```

Comment:
Did you mean to use `BaseException` here?

```suggestion
            this.errorType = ((BaseException) error).getErrorType();
```

Reply:
@sharat87 `BaseException` class doesnt have the getErrorType(). Maybe its good to leave the errorType to empty when the exception is not of `Appsmith` type.

Reply:
That may be but if the code flow has arrived at this point, that means `error instanceof AppsmithPluginException` was `false`, which means this typecast would definitely throw a `ClassCastException`. That was why I was concerned with this line. 😄